pandas/core/arrays/masked.py

```diff
@@ -957,7 +957,7 @@ def equals(self, other) -> bool:
         return array_equivalent(left, right, dtype_equal=True)
 
     def _quantile(
-        self: BaseMaskedArrayT, qs: npt.NDArray[np.float64], interpolation: str
+        self, qs: npt.NDArray[np.float64], interpolation: str
     ) -> BaseMaskedArray:
```

Comment:
Pyright didn't previously find this. I assume the return value can be any `BaseMaskedArray`.

Reply:
yep. maybe add type annotations to `_maybe_mask_result` also.

Reply:
can be follow-on. I suspect that `_maybe_mask_result` should probably return the same type, but since we use explicit Subclass constructors this would not be the case if we subclass further.

Reply:
I will leave this out - it would require quite a few ignore comments if we do something like:

```py
    @overload
    def _maybe_mask_result(self, result: ArrayLike, mask: ArrayLike) -> ArrayLike:
        ...

    @overload
    def _maybe_mask_result(
        self, result: tuple[ArrayLike, ArrayLike], mask: ArrayLike
    ) -> tuple[ArrayLike, ArrayLike]:
        ...

    def _maybe_mask_result(
        self, result: ArrayLike | tuple[ArrayLike, ArrayLike], mask: ArrayLike
    ) -> ArrayLike | tuple[ArrayLike, ArrayLike]:
```

For example `FloatingArray` and `BooleanArray` require an ndarray not an ExtensionArray.